src/common/statement_cache_manager.cpp

```diff
@@ -0,0 +1,56 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// statement_cache_manager.cpp
+//
+// Identification: src/include/common/statement_cache_manager.cpp
+//
+// Copyright (c) 2015-17, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#include "common/statement_cache_manager.h"
+
+namespace peloton {
+
+void StatementCacheManager::RegisterStatementCache(StatementCache *stmt_cache) {
+  statement_caches_.Insert(stmt_cache, stmt_cache);
+}
+
+void StatementCacheManager::UnRegisterStatementCache(StatementCache *stmt_cache) {
+  statement_caches_.Erase(stmt_cache);
+}
+
+void StatementCacheManager::InvalidateTableOid(oid_t table_id) {
+
+  if (statement_caches_.IsEmpty()) 
+    return;
+
+  // Lock the table by grabing the iterator
+  LOG_TRACE("Locking the table to get the iterator");
+  auto iterator = statement_caches_.GetIterator();
+  // Iterate each plan cache
+  for (auto &iter : iterator) {
+    iter.first->NotifyInvalidTable(table_id);
+  }
+  // Automatically release the table;
+}
+
+void StatementCacheManager::InvalidateTableOids(std::set<oid_t> &table_ids) {
```

Comment:
Is there a simple/clean way for InvalidateTableOids to call InvalidateTableOid and avoid duplication?

Reply:
InvalidateTableOids calling InvalidateTableOid would cause multiple times of mutex grabbing every time the iterator of CuckooMap is generated. I was wondering this would affect the performance.

Further, neither of these function is long or complex.

Reply:
It can stay as is. 
We don't want to add mutex acquisitions.